src/main/java/bot/utility/Parser.java

```diff
@@ -0,0 +1,143 @@
+        List<Task> tasks = list.showTasks();
+        try {
+            checkInput(words);
+            switch (words[0]) {
+            case "bye":
+                logger.write(tasks);
+                flag = 0;
+                log.append("\n\t Peace out!");
+                break;
+            case "list":
+                log.append("\n\t Here are the tasks in your list:");
+                for (int i = 0; i < tasks.size(); i++) {
+                    log.append("\n\t ").append(i + 1).append(". ").append(tasks.get(i));
+                }
+                break;
+            case "done":
+                int idx = Integer.parseInt(words[1]);
+                Task task = tasks.get(idx - 1);
+                if (task.getStatusIcon().equals("X")) {
+                    log.append("\t This task has been marked as done");
+                } else {
+                    task.markAsDone();
+                    log.append("\n\t Nice! I've marked this task as done:\n\t\t").append(task);
+                }
+                break;
+            case "find":
+                log.append("\n\t Here are the matching tasks in your list:");
+                for (int i = 0; i < tasks.size(); i++) {
+                    if (tasks.get(i).toString().contains(words[1])) {
+                        log.append("\n\t ").append(i + 1).append(". ").append(tasks.get(i));
+                    }
+                }
+                break;
+            case "todo":
+                ToDo newTask = new ToDo(words[1]);
+                tasks.add(newTask);
+                log.append("\n\t Got it. I've added this task:\n\t\t").append(newTask).append("\n\t Now you have ")
+                        .append(tasks.size()).append(" tasks in the list.");
+                break;
+            case "deadline":
+                String[] details = words[1].split(" /by ", 2);
+                Deadline deadline = new Deadline(details[0], details[1]);
+                tasks.add(deadline);
+                log.append("\n\t Got it. I've added this task:\n\t\t").append(deadline)
+                        .append("\n\t Now you have ").append(tasks.size()).append(" tasks in the list.");
+                break;
+            case "event":
+                details = words[1].split(" /at ", 2);
+                Event event = new Event(details[0], details[1]);
+                tasks.add(event);
+                log.append("\n\t Got it. I've added this task:\n\t\t").append(event)
+                        .append("\n\t Now you have ").append(tasks.size()).append(" tasks in the list.");
+                break;
+            case "delete":
+                idx = Integer.parseInt(words[1]);
+                task = tasks.get(idx - 1);
+                tasks.remove(idx - 1);
+                log.append("\n\t Noted. I've removed this task:\n\t\t").append(task)
+                        .append("\n\t Now you have ").append(tasks.size()).append(" tasks in the list.");
```

Comment:
Would it be  better for readability if you use String format instead of multiple appends?

Reply:
Will look into this, thank you!